pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -15,30 +14,53 @@ self: super:
 let
   inherit (neovimUtils) grammarToPlugin;
 
-  overrides = prev: {
+  buildQueries =
```

Comment:
Looks like moving this to separate derivations is a mistake, it bloats the runtime path, which slows down startup time significantly. For me, it is 200ms before this PR and 400ms after. If I only install parsers using `builtins.attrValues nvim-treesitter.parsers`, startup time is 120ms (although parsers without queries indeed do not work

Reply:
This seems like an issue with `nvim-treesitter.withEverything`.
It bloated the runtimepath before (which was why I never used it), it's just more noticeable now. 

My suggestion would be to have `withX` build a single derivation that symlinks all the individual queries and parsers (so they're in a single `parsers` or `queries` directory, respectively, from nvim's point of view).
But we definitely shouldn't modify the nvim-treesitter package's `$out` directory.

Reply:
> My suggestion would be to have withX build a single derivation that symlinks all the individual queries and parsers (so they're in a single parsers or queries directory, respectively, from nvim's point of view).

Someone already tried to do that, but it broke a lot of things, so it is not really possible

> I think your issue should be catppuccin related

Removing catppuccin resulted in 300ms, so not really

Reply:
> > My suggestion would be to have withX build a single derivation that symlinks all the individual queries and parsers (so they're in a single parsers or queries directory, respectively, from nvim's point of view).
> 
> Someone already tried to do that, but it broke a lot of things, so it is not really possible

What broke? nvim-treesitter installs parsers and queries like that, so I see no reason why it shouldn't be possible.

See also [my comment about nvim-treesitter planning to install queries that are bundled with parsers](https://github.com/NixOS/nixpkgs/pull/470883#discussion_r2628563001). We cannot assume that nvim-treesitter will be the source of all queries in the future.

Reply:
Oh, that sounds great. We can do that after all. But we should collate grammars & queries in the wrapper, not in `withPlugins` this time.

For now, let's bundle queries with parsers, just like we did with the old treesitter.

Reply:
Yep, startup time is now 200ms for me, thanks! Will daily drive this for a few days before approving